cgo_go122.go

```diff
@@ -0,0 +1,30 @@
+//go:build go1.22
```

Comment:
this version is not even out yet, is there some feature required from future release?

Reply:
Because I read the envoy source code, I found this CGO optimization code like this: https://github.com/envoyproxy/envoy/blob/main/contrib/golang/filters/network/source/go/pkg/network/cgo_go122.go